### PR TITLE
[ST] unused tags removal: internalclients, arm arch unsupported

### PR DESCRIPTION
**Pull request by henryZrncik**
### Type of change

- Refactoring

### Description

Removal of following Tags from System Tests: 

- **internalclients**: as these are not used for any particular reason and were created due to now obsolete reasons  which also resulted in fact that these tags were present only in fraction of all tests they were used in.
- **arm64unsupported**: removed as all test that were using it were from now on enabled also on this architecture (as of https://github.com/strimzi/test-clients/issues/75). 

note: despite removing tag internalclients we decided to keep externalclients as these seems to be used for local testing purposes.

### CODE REVIEW

File: development-docs/TESTING.md [hunk 1]
```diff
@@ -276,8 +276,6 @@ The following table shows currently used tags:
 | mirrormaker2       |                 Execute tests that deploy KafkaMirrorMaker2 resource                  |
 | conneccomponents   |  Execute tests that deploy KafkaConnect, KafkaMirrorMaker2, KafkaConnector resources  |
 | bridge             |                          Execute tests that use Kafka Bridge                          |
-| internalclients    |         Execute tests that use internal (from the Pod) Kafka clients in tests         |
-| externalclients    |          Execute tests that use external (from code) Kafka clients in tests           |
```
Comment (im-konge) [inline development-docs/TESTING.md:280]: I would personally keep the `externalclients` as I'm using it sometimes when I'm running the tests locally or on some different env when you want to try tests where we are sending/receiving messages externally (IIRC not every Nodeport or Loadbalancer related ST has external clients there).

**Review by im-konge — COMMENTED** [review]
Just one thing, but LGTM otherwise. Thanks 🚀

File: development-docs/TESTING.md (hunk 1 above)
Comment (henryZrncik) [reply development-docs/TESTING.md:280]: sounds reasonable 👍 Will keep external clients tag ... thanks for giving head up about usage!

**Review by im-konge — APPROVED** [review]

**Review by Frawless — APPROVED** [review]

**Pull request merged**